### DISCUSSION TITLE
Issue with TS in vue template tag

### DISCUSSION THREAD

**Discussion by jenky** [tighten/ziggy#746] [Q&A] [answered]
My current setup includes InertiaJS, Vue 3 and `@routes` blade directive. The `route` function works fine in `script setup` but not inside the Vue `template` tag. The app continues to function properly, but TS keeps complaining:

```
Property 'route' does not exist on type
```

I have already followed the guide in https://github.com/tighten/ziggy?tab=readme-ov-file#typescript to add the `route` declaration into `.d.ts` file.

The plugin is imported from `vendor` directory

```ts
import { ZiggyVue } from '../../vendor/tightenco/ziggy'

// ...
createApp(...)
  .use(ZiggyVue)
```

How can I make the TS shut up?

**Comment by bakerkretzmar** [comment]
Interesting, I have apps with basically this same setup and I don't remember having this issue but I'm not sure why. I'll take a look this week.

**Comment by bakerkretzmar** [comment, marked as answer]
Does adding this under your other global route declaration in you `.d.ts` fix this?

```ts
declare module 'vue' {
    interface ComponentCustomProperties {
        route: typeof routeFn;
    }
}
```

**Reply by jenky** [reply]
It works indeed. Ty so much 😄
